pkg/vm/compiler/codegen.go

```diff
@@ -37,6 +37,9 @@ type codegen struct {
 
 	// Label table for recording jump destinations.
 	l []int
+
+	// holds error of code generator.
+	Err error
```

Comment:
Do you really need it? You already have `prog.Err` that can be reused for this.

Reply:
because prog.Err holding io errors and we have others as well like having checks for "opcode %s is not a jump or call type" or  "syscall api cannot be of length 0".

Reply:
Well, I'd rather have one place to check for errors within this context. The error itself may be not related to I/O, but who cares? And you may even embed this structure in the `codegen`, so you would have unified `Err`.

Reply:
Yes, I like and idea to have one place but don't want to mixup things unrelated to each other. For me it makes code strange and not clear. Didn't get an idea how to embed.
And one more thing: if we will write to io.err some other not related errors it could appear that at some point writer can't write because of unrelated error.

Reply:
Like that:
```
type codegen struct {
    io.BufBinWriter
```
That's the question of what is more important here, semantic error separation or error context locality. I think the second one, because you can't really proceed with any kind of error here. The only theoretical problem is that this error can be somehow mangled on it's way up to the error checker, but again it's better than the possibility of just missing the check that you have now with two error variables within the same context.